File: app/src/main/java/eu/qm/fiszki/drawer/drawerItem/SelectCategory.java
package eu.qm.fiszki.drawer.drawerItem;

import android.app.Activity;

import com.mikepenz.materialdrawer.model.PrimaryDrawerItem;
import com.mikepenz.materialdrawer.model.interfaces.DescribableKt;
import com.mikepenz.materialdrawer.model.interfaces.IconableKt;
import com.mikepenz.materialdrawer.model.interfaces.NameableKt;

import java.util.ArrayList;

import eu.qm.fiszki.Alert;
import eu.qm.fiszki.R;
import eu.qm.fiszki.dialogs.category.SelectCategoryDialog;
import eu.qm.fiszki.model.category.Category;
import eu.qm.fiszki.model.category.CategoryRepository;
import eu.qm.fiszki.model.flashcard.FlashcardRepository;

/**
 * Created by Siusiacz on 09.07.2016.
 */
public class SelectCategory extends PrimaryDrawerItem {

    CategoryRepository categoryRepository;
    FlashcardRepository flashcardRepository;

    public SelectCategory(final Activity activity) {
        categoryRepository = new CategoryRepository(activity);
        flashcardRepository = new FlashcardRepository(activity);

        final ArrayList<Category> categoryToPopulate = new ArrayList<Category>();
        categoryToPopulate.add(categoryRepository.getCategoryByID(1));
        categoryToPopulate.addAll(categoryRepository.getUserCategory());

        NameableKt.withName(this, R.string.drawer_select_name);
        DescribableKt.withDescription(this, R.string.drawer_select_sub);
        IconableKt.withIcon(this, R.drawable.ic_category_select);
        this.withOnDrawerItemClickListener((view, drawerItem, position) -> {
            if (flashcardRepository.getAllFlashcards().isEmpty()) {
                new Alert().addFiszkiToFeature(activity).show();
            } else {
                new SelectCategoryDialog(activity, categoryToPopulate).show();
            }
            return false;
        });
    }
}

File: app/src/main/java/eu/qm/fiszki/drawer/drawerItem/SelectCategory.kt
package eu.qm.fiszki.drawer.drawerItem

import android.app.Activity
import com.mikepenz.materialdrawer.model.PrimaryDrawerItem
import com.mikepenz.materialdrawer.model.interfaces.withDescription
import com.mikepenz.materialdrawer.model.interfaces.withIcon
import com.mikepenz.materialdrawer.model.interfaces.withName
import eu.qm.fiszki.Alert
import eu.qm.fiszki.R
import eu.qm.fiszki.dialogs.category.SelectCategoryDialog
import eu.qm.fiszki.model.category.CategoryRepository
import eu.qm.fiszki.model.flashcard.FlashcardRepository

class SelectCategory(activity: Activity) : PrimaryDrawerItem() {

    init {
        val categoryRepository = CategoryRepository(activity)
        val flashcardRepository = FlashcardRepository(activity)

        val categoryToPopulate = ArrayList<eu.qm.fiszki.model.category.Category>().apply {
            categoryRepository.getCategoryByID(1)?.let { add(it) }
            addAll(categoryRepository.getUserCategory())
        }

        withName(R.string.drawer_select_name)
        withDescription(R.string.drawer_select_sub)
        withIcon(R.drawable.ic_category_select)
        withOnDrawerItemClickListener { _, _, _ ->
            if (flashcardRepository.getAllFlashcards().isEmpty()) {
                Alert().addFiszkiToFeature(activity).show()
            } else {
                SelectCategoryDialog(activity, categoryToPopulate).show()
            }
            false
        }
    }
}
